File: reference/native/wear-os/buildable/app/src/main/kotlin/com/goreecloud/glazeui/reference/wearable/GlazeWearableReference.kt
package com.goreecloud.glazeui.reference.wearable

import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.wear.compose.foundation.lazy.TransformingLazyColumn
import androidx.wear.compose.material3.Button
import androidx.wear.compose.material3.MaterialTheme
import androidx.wear.compose.material3.Text
import androidx.wear.compose.material3.minimumInteractiveComponentSize

/**
 * Glaze UI wearable Development Candidate reference.
 *
 * This is implementation evidence only. It is not a Stable Glaze UI contract,
 * production-ready application screen, or real-device acceptance record.
 */
@Composable
fun GlazeWearableReference(
    status: String = "All systems healthy",
    onOpenDetails: () -> Unit = {},
) {
    MaterialTheme {
        TransformingLazyColumn(modifier = Modifier.fillMaxSize()) {
            item {
                Text("Glaze UI")
            }
            item {
                Text(status)
            }
            item {
                Button(
                    onClick = onOpenDetails,
                    modifier = Modifier.minimumInteractiveComponentSize(),
                ) {
                    Text("Open details")
                }
            }
        }
    }
}

// TransformingLazyColumn provides the expected vertical wearable flow and
// current Compose for Wear OS rotary behavior. The action retains the platform
// minimum interactive component sizing rather than shrinking a phone control.

File: reference/native/watchos/GlazeWearableReference.swift
import SwiftUI

/// Glaze UI wearable Development Candidate reference.
///
/// This is implementation evidence only. It is not a Stable Glaze UI contract,
/// production-ready application screen, or real-device acceptance record.
struct GlazeWearableReference: View {
    var status: String = "All systems healthy"
    var onOpenDetails: () -> Void = {}

    var body: some View {
        ScrollView {
            VStack(alignment: .leading, spacing: 12) {
                Text("Glaze UI")
                    .font(.headline)

                Text(status)
                    .font(.body)
                    .accessibilityLabel("Status")
                    .accessibilityValue(status)

                Button("Open details", action: onOpenDetails)
                    .buttonStyle(.borderedProminent)
                    .accessibilityHint("Opens the focused detail task")
            }
            .frame(maxWidth: .infinity, alignment: .leading)
            .padding()
        }
    }
}

#Preview {
    GlazeWearableReference()
}

// ScrollView preserves the expected vertical watch interaction model and lets
// watchOS provide native Digital Crown scrolling behavior. Standard SwiftUI
// controls retain system accessibility, focus, text-scaling, and interaction
// semantics rather than recreating phone controls at watch size.
